### PR TITLE
Add macro for throttled log output

### PR DESCRIPTION
**Pull request by martiniil**

### CODE REVIEW

**Review by agutenkunst — CHANGES_REQUESTED** [review]

File: include/psen_scan_v2/logging.h [hunk 1]
```diff
@@ -25,11 +26,31 @@
     console_bridge::getOutputHandler()->log(fmt::format("{}: {}", name, fmt::format(__VA_ARGS__)), level, file, line); \
   }
 
+#define PSENSCAN_LOG_THROTTLE(period, name, file, line, level, ...)                                                    \
+  {                                                                                                                    \
+    static std::chrono::system_clock::time_point throttle_last_hit;                                                    \
```
[the remark this answers is not in the document]
Comment (martiniil) [reply include/psen_scan_v2/logging.h:31]: Note that this is a macro. Every usage is unique!

[the remark this answers is not in the document]
Comment (agutenkunst) [reply include/psen_scan_v2/logging.h:31]: 1. Wrong - Macro is preprocessor. The static is global.
2. Proof to me by test

[the remark this answers is not in the document]
Comment (martiniil) [reply include/psen_scan_v2/logging.h:31]: You have a local scope inside the macro ...

File: include/psen_scan_v2/logging.h [hunk 2]
```diff
@@ -16,14 +16,31 @@
...
 
 #define PSENSCAN_LOG(name, file, line, level, ...)                                                                     \
+  do                                                                                                                   \
   {                                                                                                                    \
     console_bridge::getOutputHandler()->log(fmt::format("{}: {}", name, fmt::format(__VA_ARGS__)), level, file, line); \
-  }
+  } while (false)  // https://stackoverflow.com/questions/1067226/c-multi-line-macro-do-while0-vs-scope-block
+
+#define PSENSCAN_LOG_THROTTLE(period, name, file, line, level, ...)                                                    \
+  PSENSCAN_LOG_THROTTLE_INTERNAL(std::chrono::system_clock::now, period, name, file, line, level, __VA_ARGS__)
```
Comment (agutenkunst) [inline include/psen_scan_v2/logging.h:31]:
```suggestion
  PSENSCAN_LOG_THROTTLE_INTERNAL(std::chrono::system_clock::now(), period, name, file, line, level, __VA_ARGS__)
```

Comment (agutenkunst) [reply include/psen_scan_v2/logging.h:31]: Rest accordingly. Should simplify things...

**Comment by agutenkunst** [comment]
@martiniil I added some changes. Please check before merge.

**Pull request merged**